checkov/terraform/checks/graph_checks/aws/CLoudFrontS3OriginConfigWithOAI.yaml

```diff
@@ -0,0 +1,30 @@
+metadata:
+  id: "CKV2_AWS_46"
+  name: "Ensure AWS Cloudfront Distribution with S3 have Origin Access set to enabled"
+  category: "IAM"
+definition:
+  or:
+    - and:
+        - cond_type: filter
+          attribute: resource_type
+          operator: within
+          value:
+            - aws_cloudfront_distribution
+        - cond_type: connection
+          resource_types:
+            - aws_cloudfront_distribution
+          connected_resource_types:
+            - aws_s3_bucket
+          operator: not_exists
+    - cond_type: attribute
+      resource_types:
+        - aws_cloudfront_distribution
+      attribute: "origin.*.s3_origin_config"
+      operator: "exists"
```

Comment:
Shouldn't this be `not_exists`???

Reply:
Will add a test and file a PR ... and while there also fix the typo in the file name :)